packages/portal/backend/src/Factory.ts

```diff
@@ -76,24 +82,46 @@ export class Factory {
             Log.trace("Factory::getCourseController() - using provided controller");
         }
         try {
+            const pluginPath = Config.getInstance().getProp(ConfigKey.plugin_fullpath);
+
+            Log.trace("Factory::getCourseController() - name: " + name + "; path: " + pluginPath);
+
             // NOTE: using require instead of import because file might not be present in forks
             // import complains about this, but require does not.
-            let plug: any;
-            if (name === 'classytest') {
-                Log.trace("Factory::getCourseController() - name: " + name + " - plug: DefaultCourseController");
-                plug = await require('./custom/DefaultCourseController'); // default for testing
-            } else {
-                // If a course wants to specialize the AdminView it should be in the file below.
-                // This is not required. But if it is added, it should never be pushed back to 'classy/master'
-                Log.trace("Factory::getCourseController() - name: " + name + " - plug: CustomCourseController");
-                plug = await require('./custom/CustomCourseController');
+            let plug: any = null;
+
+            try {
+                if (name !== Factory.TESTNAME && typeof pluginPath !== "undefined") {
+                    // If a course wants to specialize the AdminView it should be in the file below.
+                    // This is not required. But if it is added, it should never be pushed back to 'classy/master'
+                    Log.trace("Factory::getCourseController() - name: " + name + " - plug: CustomCourseController");
+                    plug = await require(pluginPath + "/plugin/src/backend/CustomCourseController");
+                }
+            } catch (err) {
+                const msg = "Factory::getCourseController() - src/custom/CustomCourseController.ts must be defined";
+                Log.error(msg);
+                Log.error(err);
+                plug = null;
+            }
+
+            if (plug === null) {
+                // either it's a test, or the previous step did not finish due to error or no defined CustomCourseController
+                if (name === Factory.TESTNAME || typeof pluginPath === "undefined") {
+                    Log.trace("Factory::getCourseController() - name: " + name + " - plug: DefaultCourseController");
+                    plug = await require("./custom/DefaultCourseController"); // default for testing
+                } else {
+                    // If a course wants to specialize the AdminView it should be in the file below.
+                    // This is not required. But if it is added, it should never be pushed back to 'classy/master'
+                    Log.trace("Factory::getCourseController() - name: " + name + " - plug: CustomCourseController");
+                    plug = await require(pluginPath + "/plugin/src/backend/CustomCourseController");
```

Comment:
Which one should it be? 

`"/plugin/src/backend/CustomCourseController"`
or 
![image](https://user-images.githubusercontent.com/7872295/116107614-08d63e00-a668-11eb-946d-9376297f0efa.png)
Source: https://github.com/SECapstone/classy-plugin

Reply:
(Not as big of a concern, as it is a different repo than this PR but I thought you should know)